ngraph_bridge/ngraph_executor.h

```diff
@@ -47,36 +48,33 @@ class NGraphExecutor {
   // Calls Compute Signature and gets ngraph executable
   // Update the cache and if called again with the same input shapes,
   // return fromm the cache
-  Status GetNgExecutable(const std::vector<Tensor>& tf_input_tensors,
-                         std::shared_ptr<ngraph::runtime::Executable>& ng_exec,
-                         bool& cache_hit);
-
-  Status GetNgFunction(
-      const std::shared_ptr<ngraph::runtime::Executable>& ng_exec,
-      std::shared_ptr<ngraph::Function>& ng_function);
+  Status GetExecutableAndTensors(
+      const std::vector<Tensor>& tf_input_tensors,
+      std::shared_ptr<ngraph::runtime::Executable>& ng_exec,
+      shared_ptr<PipelinedTensorsStore>& pts, bool& cache_hit);
 
   // TODO Rename this to DecodeAttributes
   Status ParseNodeAttributes(
       const google::protobuf::Map<string, AttrValue>& additional_attributes,
       std::unordered_map<std::string, std::string>* additional_attribute_map);
 
-  // Returns OK when a set of i/o tensor is available. Returns error message
-  // otherwse.
-  // The caller can wait or come back later - based on what the application
-  // demands
-  Status GetTensorsFromPipeline(
-      const std::shared_ptr<ngraph::runtime::Executable>& ng_exec,
-      std::tuple<int, PipelinedTensorVector, PipelinedTensorVector>&
-          io_tensors);
-
-  void ReturnPipelinedTensors(
-      std::shared_ptr<ngraph::runtime::Executable> ng_exec, size_t idx) {
-    lock_guard<mutex> lock(m_mutex);
-    m_executable_pipelined_tensors_map.at(ng_exec)->return_tensors(idx);
-  }
+  // Callback function called from NgraphDataCache's LookUpOrCreateItem() method
+  // Creates ng_executable, serialized_ng_function, and initializes I/O
+  // TensorPipeline
+  std::pair<Status, std::tuple<std::shared_ptr<ngraph::runtime::Executable>,
+                               std::string, shared_ptr<PipelinedTensorsStore>>>
+  CreateCallback(std::string signature, std::vector<TensorShape> input_shapes,
```

Comment:
Rename this function. It sounds generic when it is not

Reply:
Please suggest some name